README.md

```diff
@@ -186,4 +186,91 @@ backend (e.g Jaeger Thrift Span), and then push them to corresponding backend or
 
 ## OpenCensus Collector
 
-TODO: add content about Collector.
+The OpenCensus Collector is a component that runs “nearby” (e.g. in the same
+VPC, AZ, etc.) a user’s application components and receives trace spans and
+metrics emitted by tasks instrumented with opencensus instrumentation (or other
+supported instrumentation protocols/libraries). The received spans and metrics
+could be emitted directly by clients in instrumented tasks, or potentially
+routed via intermediate proxy sidecar/daemon agents (such as opencensus-agent).
+The collector provides a central egress point for exporting traces and metrics
+to one or more tracing and metrics backends, with buffering and retries as well
+as advanced aggregation, filtering and annotation capabilities.
+
+The collector is extensible enabling it to support a range of out-of-the-box
+(and custom) capabilities such as:
+
+* Retroactive (tail-based) sampling of traces
+* Cluster-wide z-pages
+* Filtering of traces and metrics
+* Aggregation of traces and metrics
+* Decoration with meta-data from infrastructure provider (e.g. k8s master)
+* much more ...
+
+The collector also serves as a control plane for agents/clients by supplying
+them updated configuration (e.g. trace sampling policies), and reporting
+agent/client health information/inventory metadata to downstream exporters.
+
+### Architecture Overview
+
+The OpenCensus Collector runs as a standalone instance and receives spans and
+metrics exporterd by one or more OpenCensus Clients or Agents, or by
```

Comment:
nit: Please replace "Client" with "Library", to be consistent with the previous section.